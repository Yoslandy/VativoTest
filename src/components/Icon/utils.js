export const icons = {
    success: "verified_user",
    warning: "warning",
    settings: "settings",
    add_circle_outline: "add_circle_outline",
    eye: "visibility",
    time: "access_time",
    save: "save_outline",
    download: "file_download_outline",
    delete: "delete_outline",
    reset: "key_outline",
    close: "close_outline",
    "eye-slash": "visibility_off",
    error: "do_disturb_alt",
    error_alt: "block_outline",
    login: "login",
    filter_alt: "filter_alt",
    search: "search",
    double_arrow: "keyboard_double_arrow_right",
    arrow: "keyboard_arrow_right",
    queue: "queue",
    show_chart: "show_chart",
    people_alt: "people_alt",
    help_outline: "help_outline",
    account_circle: "account_circle",
    person: "person",
    apartment: "apartment",
    arrow_downward: "arrow_downward",
    arrow_upward: "arrow_upward",
    person_outline: "person_outline",
}

export const styles = {
    outline: "material-icons-outlined",
    default: "material-icons",
}
