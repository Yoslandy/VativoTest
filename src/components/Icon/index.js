import React from "react"
import PropTypes from "prop-types"
import { icons, styles } from "./utils"

export default function Icon(props) {
    const { type, onClick, className, appearance } = props
    return (
        <span
            className={`${styles[appearance]} ${className}`}
            onClick={onClick}
        >
            {icons[type]}
        </span>
    )
}

Icon.propTypes = {
    type: PropTypes.string,
    onClick: PropTypes.func,
    className: PropTypes.string,
    appearance: PropTypes.string,
}

Icon.defaultProps = {
    type: "eye",
    onClick: () => {},
    className: "",
    appearance: "default",
}
