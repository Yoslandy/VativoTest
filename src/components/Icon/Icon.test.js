import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import Icon from "./index"

test("Renders Icon correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <Icon type="eye" />
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
