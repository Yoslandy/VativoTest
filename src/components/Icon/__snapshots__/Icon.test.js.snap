// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Icon correctly 1`] = `
<span
  className="material-icons "
  onClick={[Function]}
  role="button"
>
  visibility
</span>
`;
